rollouts/controllers/rollout_controller.go

```diff
@@ -273,45 +274,85 @@ func (r *RolloutReconciler) computeTargets(ctx context.Context,
 }
 
 func (r *RolloutReconciler) rolloutTargets(ctx context.Context, rollout *gitopsv1alpha1.Rollout, targets *Targets) ([]gitopsv1alpha1.ClusterStatus, error) {
-
 	clusterStatuses := []gitopsv1alpha1.ClusterStatus{}
 
-	if rollout.Spec.Strategy.Type != gitopsv1alpha1.AllAtOnce {
+	if rollout.Spec.Strategy.Type != gitopsv1alpha1.AllAtOnce && rollout.Spec.Strategy.Type != gitopsv1alpha1.RollingUpdate {
 		return clusterStatuses, fmt.Errorf("%v strategy not supported yet", rollout.Spec.Strategy.Type)
 	}
 
+	concurrentUpgrades := 0
+	maxConcurrent := math.MaxInt
+
+	for _, target := range targets.Unchanged {
+		if !isRRSSynced(target) {
+			concurrentUpgrades++
+		}
+	}
+
+	if rollout.Spec.Strategy.Type == gitopsv1alpha1.RollingUpdate {
+		maxConcurrent = int(rollout.Spec.Strategy.RollingUpdate.MaxConcurrent)
+	}
+
```

Comment:
I see you caught this slightly later in your review (https://github.com/GoogleContainerTools/kpt/pull/3714#discussion_r1063034484), but to confirm, since we are creating the status from scratch here, we still need to loop through all the rollout resources here.